File: src/main/java/edu/projetJava/tools/FixFullScreen.java
package edu.projetJava.tools;

import java.io.File;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.ArrayList;
import java.util.List;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class FixFullScreen {
    public static void main(String[] args) {
        File dir = new File("src/main/java/edu/projetJava/controllers");
        if (!dir.exists()) return;
        
        Pattern pattern = Pattern.compile("^(\\s+)(stage|window)\\.setScene\\(.*");

        for (File file : dir.listFiles()) {
            if (file.getName().endsWith(".java")) {
                try {
                    List<String> lines = Files.readAllLines(Paths.get(file.toURI()));
                    List<String> newLines = new ArrayList<>();
                    boolean modified = false;
                    
                    for (int i = 0; i < lines.size(); i++) {
                        String line = lines.get(i);
                        newLines.add(line);
                        
                        Matcher matcher = pattern.matcher(line);
                        if (matcher.matches()) {
                            String indent = matcher.group(1);
                            String stageVar = matcher.group(2);
                            
                            String nextLine = (i + 1 < lines.size()) ? lines.get(i + 1) : "";
                            if (!nextLine.contains("setFullScreen")) {
                                newLines.add(indent + stageVar + ".setFullScreen(true);");
                                modified = true;
                            }
                        }
                    }
                    
                    if (modified) {
                        Files.write(Paths.get(file.toURI()), newLines);
                        System.out.println("Fixed " + file.getName());
                    }
                } catch (Exception e) {
                    e.printStackTrace();
                }
            }
        }
    }
}

File: fix_fullscreen.py
import os
import re

directory = r'c:\Users\ilyes\Downloads\Workshop-JDBC-JavaFX\src\main\java\edu\projetJava\controllers'

def fix_fullscreen():
    for filename in os.listdir(directory):
        if not filename.endswith('.java'):
            continue
        filepath = os.path.join(directory, filename)
        with open(filepath, 'r', encoding='utf-8') as f:
            lines = f.readlines()
        
        modified = False
        new_lines = []
        i = 0
        while i < len(lines):
            line = lines[i]
            new_lines.append(line)
            
            # Match stage.setScene(...) but not alertStage.setScene(...) to avoid making alerts fullscreen
            match = re.search(r'^(\s+)(stage|window)\.setScene\(', line)
            if match:
                indent = match.group(1)
                stage_var = match.group(2)
                # check next line
                next_line = lines[i+1] if i+1 < len(lines) else ""
                if "setFullScreen" not in next_line:
                    new_lines.append(f"{indent}{stage_var}.setFullScreen(true);\n")
                    modified = True
            i += 1
            
        if modified:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.writelines(new_lines)
            print(f"Updated {filename}")

fix_fullscreen()
